twisted-infra/braid
# [#203] Use a simple tox builder for pyflakes.

Scope
=====

This branch was targeted to updating the pyflakes builder to use a simple venv and to not look at the errors from previous build from trunk.

This should fix #203

Chagnes
========

It looks like the pyflakes builder was already updated, but the supporting  code was not removed.
I have removed it now.

I have created a new generic ToxBuilder factory class which will receive the tox env name ... so that we will not have to create new classes for each new builder type.

I have clean the buildmaster and buildslave code for pyflakes errors.

I have updated the buildmaster and buildslave code to work with the Vagrant VM and I have updated the documentation.


I could not find a way to redirect tox errors to stderr so that the errors can be presented into a separate stream... I find this stupid, but I hope that there is a way to fix it


How to test
=========

Start the buildbot and buildslave on vagrant


```
fab config.vagrant buildbot.install
fab config.vagrant buildbot.start
fab config.vagrant buildslave.install
fab config.vagrant buildslave.start
```

Go to http://172.16.255.140:8080/builders/static-checkers and login

trigger a build for trunk ... or another branch. check that all is ok.

I have created the `static-checkers-test` branch on Twisted main repo for testing.
